packages/sanity/src/core/form/members/object/fields/ObjectField.tsx

```diff
@@ -63,13 +65,43 @@ export const ObjectField = function ObjectField(props: {
 
   const handleChange = useCallback(
     (event: PatchEvent | PatchArg) => {
+      const isRoot = member.field.path.length === 0
+      const patches = PatchEvent.from(event).patches
+      // if the patch is an unset patch that targets an item in the array (as opposed to unsetting a field somewhere deeper)
+      const isRemovingLastItem = patches.some(
+        (patch) => patch.type === 'unset' && patch.path.length === 1
+      )
+
+      // this handle touches on more than just object fields, documents included
+      // if we're at a "document" level, then we want to have a way to skip the following logic
+      if (!isRoot) {
+        if (isRemovingLastItem) {
+          // apply the patch to the current value
+          const result = applyAll(member.field.value || {}, patches)
+
+          // only run this if the result is an empty object with only _type
+          if (result && result._type && Object.keys(result).length === 1) {
+            // The value has a _type key, but the type name from schema is 'object',
+            // but _type: 'object' is implicit so we should fix it by removing it
```

Comment:
I’m not sure I see how this comment relates here, since we’re not checking for the value of the type - but don’t let that block the PR. Would be nice to clarify the purpose of this path after merging though.

Reply:
This comment was originally one that we had in the old code of v2. 
Let me clarify it though :)

Reply:
I've updated it, hopefully it's a bit clearer why it exists 👍